src/main/java/randoop/generation/ConstantMiningSelection.java

```diff
@@ -0,0 +1,112 @@
+   * are "mined" by the {@code randoop.reflection.ClassLiteralExtractor}. These weights are never
+   * changed once initialized.
+   */
+  private final Map<Sequence, Double> literalWeightMap = new HashMap<>();
+
+  /**
+   * Initialize GRT Constant Mining selection by computing weights for literals that appear in
+   * classes under test.
+   *
+   * @param componentManager component generator from {@link ForwardGenerator} used for getting the
+   *     frequency of a literal
+   * @param numClasses number of classes under test
+   * @param literalTermFrequencies a map from a literal to the number of times it appears in any
+   *     class under test
+   */
+  public ConstantMiningSelection(
+      ComponentManager componentManager,
+      int numClasses,
+      Map<Sequence, Integer> literalTermFrequencies) {
+    Map<Sequence, Integer> literalDocumentFrequencies =
+        componentManager.getLiteralDocumentFrequency();
+
+    if (GenInputsAbstract.constant_mining_logging) {
+      System.out.println("Literal term frequencies: ");
+      System.out.println(literalTermFrequencies);
+      System.out.println("Literal document frequencies: ");
+      System.out.println(literalDocumentFrequencies);
+    }
+
+    // Iterate through all literals that were found by the ClassLiteralExtractor.
+    for (Sequence sequence : literalDocumentFrequencies.keySet()) {
+      Integer documentFrequency = literalDocumentFrequencies.get(sequence);
+      Integer termFrequency = literalTermFrequencies.get(sequence);
+
+      // Compute the term frequency-inverse document frequency for GRT Constant Mining.
+      double tfIdf =
+          termFrequency * Math.log((numClasses + 1.0) / ((numClasses + 1.0) - documentFrequency));
+      literalWeightMap.put(sequence, tfIdf);
+    }
+  }
+
+  /**
+   * Select a sequence from the candidate list. The candidate list contains input sequences that
+   * produce a specific type.
+   *
+   * <p>The {@code literalWeightMap} is concerned only with input sequences that are literals
+   * extracted from the classes under test. Other input sequences that are generated by Randoop
+   * throughout the test generation process will not have a corresponding weight in the
+   * literalWeightMap; instead, we use an arbitrary value of 1. (The GRT paper does not say what to
+   * do in that case.)
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    if (GenInputsAbstract.constant_mining_logging) {
+      outputCandidateWeights(candidates);
+    }
+
+    // Iterate through the candidate list and assign any sequences with no weight a weight of 1.
+    for (int i = 0; i < candidates.size(); i++) {
+      Sequence candidate = candidates.get(i);
+      if (!literalWeightMap.containsKey(candidate)) {
+        // TODO: Using the value 1.0 seems questionable.
+        literalWeightMap.put(candidate, 1.0);
```

Comment:
Yeah, this can heavily skew input selection. This makes it very difficult to replicate GRT.

Perhaps we can pursue one of these ideas:
* Set a fixed probability to choose non-Randoop generated sequences that grows over time (slowing down over time)
* Somehow represent the term frequency-inverse document frequency of non-Randoop generated sequences. We have the number of classes in the SUT, but we do not have the number of classes "containing" the "constant" $t$: $|d \in D: t \in d|$. Perhaps an argument is that the number of such classes is small, so this frequency-inverse document frequency is also small, and thus 1.0 is a good approximation for a weighting.